### PR TITLE
refactor(app): update H-S TestShake slideout and overflow menu

### LINKED ISSUE #11045: 6.0 Feedback:  Wrong tooltip is shown when cursor is over stop button in heater shake test shake slide out
### Overview

When in the test shake slide out menu if you have over the stop button the tool tip for open latch shows 

### Current Behavior

Open tool tip shows under stop shaking button

### Expected Behavior

Stop tool tip shows

### Steps To Reproduce

1. Open App
2. Go to robot with H/S
3. Click overflow menu
4. set shake speed
5. Go to test shake
6. Hoover sto
Screen Shot 2022-07-07 at 1 44 54 PM
Screen Shot 2022-07-07 at 1 45 04 PM

https://user-images.githubusercontent.com/106115893/177837327-4c658344-35c7-4ddd-8ea5-450c3f1c301b.mov

p

### Operating system

Mac

### Robot setup or anything else?

_No response_

### LINKED ISSUE #11074: Update 'test shake' slide-out & overflow menu selection
# Overview

Through testing, we found redundancy in H-S dropdown selection options.  Our path forward will be to use the "test shake" selection to offer users the ability to use the shaker outside of a protocol for purposes of testing, fine tuning, or starting a shake prior to a protocol. 

# Implementation Details
- Remove "set shake speed" dropdown selection and slideout
- If shake has been started via 'test shake' slideout, dropdown selection should update to read "Deactivate shaker.' clicking this this will stop the shaker 
- Labware latch control CTA updated to read "Open / Close latch." Functionality remains the same (open/close latch accordingly; when latch is open CTA is to close and vice versa)
- Troubleshooting section is removed from current design
- Module card is removed from test shake slideout
- Add link to show attachment instructions
- "Start" is disabled if the latch is open. Show a message on hover over the start button that reads "cannot shake when labware latch is open"
- "Open latch" is disabled if the module is shaking. Show a message on hover over the open latch button that reads "Cannot open latch while module is shaking"
- Overflow menu updates apply to the devices tab as well as the run detail page

# Design links
- Test shake slideout: https://www.figma.com/file/695ogls23YsQYImRcGYDE1/Heater-Shaker?node-id=5610%3A94457
- Overflow menu: https://www.figma.com/file/695ogls23YsQYImRcGYDE1/Heater-Shaker?node-id=5610%3A94459

# Acceptance Criteria
- [ ] "Set shake speed" no longer displays in the dropdown selection
- [ ] If shake has been started via 'test shake' slideout, dropdown selection should update to read "deactivate shaker' - clicking this this will stop the shaker 
- [ ] Labware latch control button in the test shake slideout CTA reads "Open / Close latch." Functionality remains the same (open/close latch accordingly; when latch is open CTA is to close and vice versa)
- [ ] Troubleshooting section no longer displays in Test Shake slideout
- [ ] Module card no longer displays in test shake slideout
- [ ] Link to "Show attachment instructions" displays in test shake slideout. Clicking this opens the set up wizard
- [ ] "Start" is disabled if the latch is open. Show a message on hover over the start button that reads "cannot shake when labware latch is open" 
- [ ] Shake CTA reads "Stop" if module is shaking. Clicking this stops shake (this functionality is already in place)
- [ ] "Open latch" is disabled if the module is shaking. Show a message on hover over the open latch button that reads "Cannot open latch while module is shaking"
- [ ] User can start a shake via the "Test Shake" slideout. If slideout is closed without stopping shake, shake continues on the module. If a protocol is starting while the module is shaking, user will see a modal pop up asking if user would like to continue or stop shake
- [ ] overflow menu updates in this ticket are applied to both the devices & run detail page menus

### LINKED ISSUE #11136: Move latch state to below the title in test shake slideout
# Overview

In the test shake slideout, move the state of the latch from on the "labware latch" title line to below it.

# Implementation Details
- Move latch status (Open / Closed) to below the "Labware Latch" title 

# Design Links
- https://www.figma.com/file/695ogls23YsQYImRcGYDE1/6.1-Heater%2FShaker?node-id=2841%3A104034

# Acceptance Criteria
- [x] Current status of labware latch (open/closed) appears below the Labware Latch title in the test shake slideout

### PR DESCRIPTION
**Pull request by sakibh**
# Overview

This PR updates the TestShake slideout and updates the overflow menu selection. It also moves the labware latch status below the title. Additionally, it addresses the tooltip bug on the slideout. closes #11074 closes #11136 closes #11045

Screen Shot 2022-07-18 at 2 27 01 PM

image

Screen Shot 2022-07-18 at 2 27 26 PM

# Changelog

- Remove module card from TestShake slideout.
- Remove troubleshooting section.
- Update button copy
- Remove `Set shake speed` from overflow menu
- Conditionally show `Deactivate shaker` in the menu when test shake is active.
- Removed set shake logic from slideout
- Fix tooltip bug
- Move latch status below the title

# Review requests

- Check that `Set shake speed` no longer shows in the overflow menu
- Start a test shake and close the slideout. Opening the overflow menu should now show `Deactivate shaker`. Clicking this should stop shaking.
- In the TestShake slideout the labware latch control button should read `Open latch` and `Close latch`. Functionality should persist as before.
- Troubleshooting section should no longer be displayed.
- Clicking `Show attachment instructions` should open the setup wizard.
- "Start" should be disabled if the latch is open. It should show a message on hover over the start button that reads "cannot shake when labware latch is open".
- Shake CTA reads "Stop" if module is shaking. Clicking this stops shake.
- "Open latch" should be disabled if the module is shaking. It should show a message on hover over the open latch button that reads "Cannot open latch while module is shaking".
- Start a test shake via the slideout. If slideout is closed without stopping shake, shake continues on the module. Run a protocol. You should see a modal pop up asking if user would like to continue or stop shake.
- The overflow menu and TestShake slideout updates should reflect on both the Devices and Run Detail Page menus.
- Check that the labware latch status is below the title
- Check that the tooltip bug is fixed

# Risk assessment

low

### CODE REVIEW

File: app/src/organisms/ModuleCard/TestShakeSlideout.tsx [hunk 1]
```diff
@@ -118,6 +114,22 @@ export const TestShakeSlideout = (
       ? t('input_out_of_range', { ns: 'device_details' })
       : null
 
+  const getLatchStatus = (latchStatus: LatchStatus): string => {
+    switch (latchStatus) {
+      case 'opening':
+      case 'idle_open':
+      case 'idle_unknown': {
+        return t('open', { ns: 'shared' })
```
Comment (jerader) [inline app/src/organisms/ModuleCard/TestShakeSlideout.tsx:135]:
```suggestion
        return t('shared:open')
```

File: app/src/organisms/ModuleCard/TestShakeSlideout.tsx [hunk 2]
```diff
@@ -118,6 +114,22 @@ export const TestShakeSlideout = (
...
+      }
+      case 'closing':
+      case 'idle_closed': {
+        return t('closed', { ns: 'heater_shaker' })
```
Comment (jerader) [inline app/src/organisms/ModuleCard/TestShakeSlideout.tsx]:
```suggestion
        return t('heater_shaker:closed')
```

File: app/src/organisms/ModuleCard/TestShakeSlideout.tsx [hunk 3]
```diff
@@ -159,36 +171,27 @@ export const TestShakeSlideout = (
...
             <Text
-              fontSize={TYPOGRAPHY.fontSizeP}
+              textTransform={TEXT_TRANSFORM_CAPITALIZE}
+              fontSize={TYPOGRAPHY.fontSizeLabel}
               fontWeight={TYPOGRAPHY.fontWeightSemiBold}
               color={COLORS.darkBlack}
             >
-              {t('labware_latch', { ns: 'heater_shaker' })}
+              {t('labware_latch_status', {
+                ns: 'heater_shaker',
```
Comment (jerader) [inline app/src/organisms/ModuleCard/TestShakeSlideout.tsx]: same comment as above

File: app/src/organisms/ModuleCard/TestShakeSlideout.tsx [hunk 4]
```diff
@@ -202,8 +205,8 @@ export const TestShakeSlideout = (
             {...targetProps}
           >
             {!isLatchClosed
-              ? t('close', { ns: 'shared' })
-              : t('open', { ns: 'shared' })}
+              ? t('close_latch', { ns: 'heater_shaker' })
+              : t('open_latch', { ns: 'heater_shaker' })}
```
Comment (jerader) [inline app/src/organisms/ModuleCard/TestShakeSlideout.tsx]: same as above

File: app/src/organisms/ModuleCard/TestShakeSlideout.tsx [hunk 5]
```diff
@@ -264,51 +267,19 @@ export const TestShakeSlideout = (
...
-            fontWeight={TYPOGRAPHY.fontWeightSemiBold}
-            color={COLORS.blue}
-            id={'HeaterShaker_Attachment_Instructions'}
-            onClick={() => setShowWizard(true)}
-          >
-            {t('go_to_attachment_instructions', { ns: 'heater_shaker' })}
-          </Link>
-        </Collapsible>
-      </Flex>
+        {t('show_attachment_instructions', { ns: 'heater_shaker' })}
```
Comment (jerader) [inline app/src/organisms/ModuleCard/TestShakeSlideout.tsx]: same as above

File: app/src/organisms/ModuleCard/__tests__/TestShakeSlideout.test.tsx [hunk 6]
```diff
@@ -173,16 +158,10 @@ describe('TestShakeSlideout', () => {
...
     const { getByText } = render(props)
 
-    const troubleshooting = getByText('Troubleshooting')
-    fireEvent.click(troubleshooting)
-
-    getByText(
-      'Revisit instructions for attaching the module to the deck as well as attaching the thermal adapter.'
-    )
-    getByText('Go to attachment instructions')
+    getByText('Show attachment instructions')
```
Comment (jerader) [inline app/src/organisms/ModuleCard/__tests__/TestShakeSlideout.test.tsx]: can you extend this slightly by testing clicking on the link?

**Review by jerader — COMMENTED** [review]
Looking good so far, left a few comments. Don't forget to remove all the code in the heater shaker slideout regarding set shake.

**Comment by jerader** [comment]
I have a thought regarding the `Show attachment instructions` link. It basically does the same thing that the overflow menu button `See how to attach to deck` does. Can we call these the same thing for consistency? cc @ecormany @emilywools 

Screen Shot 2022-07-14 at 12 23 23 PM
Screen Shot 2022-07-14 at 12 23 57 PM

File: app/src/organisms/ModuleCard/TestShakeSlideout.tsx [hunk 7]
```diff
@@ -264,51 +267,19 @@ export const TestShakeSlideout = (
...
-        paddingY={SPACING.spacing4}
-        width="100%"
+      {showWizard && (
+        <HeaterShakerWizard onCloseClick={() => setShowWizard(false)} />
+      )}
+      <Link
+        marginTop={SPACING.spacing2}
+        fontSize={TYPOGRAPHY.fontSizeP}
+        fontWeight={TYPOGRAPHY.fontWeightSemiBold}
+        color={COLORS.blue}
```
Comment (jerader) [inline app/src/organisms/ModuleCard/TestShakeSlideout.tsx:328]: We might have to wait until `release_6.0.0` merges into `edge` before implementing this change. But We added something in the release branch that makes the all the `links` tab-able by adding a `role="button"` and then adding a `tabIndex` to `Link`

```suggestion
      <Link
        role="button"
        marginTop={SPACING.spacing2}
         css={TYPOGRAPHY.pSemibold}
        color={COLORS.blue}
```

**Comment by emilywools** [comment]
Jethary, good call!!! i think we should match, but let's let Ed confirm

File: app/src/organisms/ModuleCard/TestShakeSlideout.tsx (hunk 7 above)
Comment (shlokamin) [reply app/src/organisms/ModuleCard/TestShakeSlideout.tsx:328]: oof, good call @jerader. i hope that rebase isn't going to be too messy for this branch

**Review by shlokamin — APPROVED** [review]
LGTM, let's def delete the set shake code in the HS slideout component. 

If there are changes to the test shake slideout in the release branch (which i did not know about), we should probably wait to merge this. I think otherwise it'll make the (already hellish) merge conflicts into edge even worse.

**Comment by ecormany** [comment]
I agree, `Show attachment instructions` is the clearer of the two phrasings, and it works in both contexts (link and menu item). Not urgent by any means, so proceed however/whenever is necessary to avoid conflicts.

**Pull request merged**